File: backend/src/modules/evaluation-models/domain/evaluation-model.ts
export type EvaluationModelItem = {
  id: string;
  teacherId: string;
  evaluationModelId: string;
  name: string;
  maxScore: number;
  sortOrder: number;
  isRecovery: boolean;
  /** Quando isRecovery, aponta para o item regular recuperado. */
  recoversItemId: string | null;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date | null;
};

export type EvaluationModel = {
  id: string;
  teacherId: string;
  name: string;
  description: string | null;
  isActive: boolean;
  sortOrder: number;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date | null;
  items: EvaluationModelItem[];
};

File: frontend/lib/domain/entities/evaluation_model.dart
class EvaluationModelItem {
  const EvaluationModelItem({
    required this.id,
    required this.evaluationModelId,
    required this.name,
    required this.maxScore,
    required this.sortOrder,
    this.isRecovery = false,
    this.recoversItemId,
  });

  final String id;
  final String evaluationModelId;
  final String name;
  final double maxScore;
  final int sortOrder;
  final bool isRecovery;
  final String? recoversItemId;
}

class EvaluationModel {
  const EvaluationModel({
    required this.id,
    required this.name,
    this.description,
    required this.isActive,
    required this.sortOrder,
    required this.items,
  });

  final String id;
  final String name;
  final String? description;
  final bool isActive;
  final int sortOrder;
  final List<EvaluationModelItem> items;
}
